docs/sources/flow/reference/components/discovery.docker.md

```diff
@@ -0,0 +1,167 @@
+[tls_config]: #tls_config-block
+
+### filter block
+
+The `filter` block configures a filter to pass to the Docker Engine to limit
+the amount of containers returned. The `filter` block can be specified multiple
+times to provide more than one filter.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`name` | `string` | Filter name to use. | | yes
+`values` | `list(string)` | Values to pass to the filter. | | yes
+
+Refer to [List containers][List containers] from the Docker Engine API
+documentation for the list of supported filters and their meaning.
+
+[List containers]: https://docs.docker.com/engine/api/v1.41/#tag/Container/operation/ContainerList
+
+### http_client_config block
+
+The `http_client_config` block configures settings used to connect to the
+Docker Engine API server.
+
+{{< docs/shared lookup="flow/reference/components/http-client-config-block.md" source="agent" >}}
+
+### basic_auth block
+
+{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" >}}
+
+### authorization block
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}
+
+### oauth2 block
+
+{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" >}}
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`targets` | `list(map(string))` | The set of targets discovered from the docker API.
+
+Each target includes the following labels:
+
+* `__meta_docker_container_id`: ID of the container.
+* `__meta_docker_container_name`: Name of the container.
+* `__meta_docker_container_network_mode`: Network mode of the container.
+* `__meta_docker_container_label_<labelname>`: Each label from the container.
+* `__meta_docker_network_id`: ID of the Docker network the container is in.
+* `__meta_docker_network_name`: Name of the Docker network the container is in.
+* `__meta_docker_network_ingress`: Set to `true` if the Docker network the is
+  an ingress network.
+* `__meta_docker_network_internal`: Set to `true` if the Docker network is an
+  internal network.
+* `__meta_docker_network_label_<labelname>`: Each label from the network the
+  container is in.
+* `__meta_docker_network_scope`: The scope of the network the container is in.
+* `__meta_docker_network_ip`: The IP of the container in the network.
+* `__meta_docker_port_private`: The private port on the container.
+* `__meta_docker_port_public`: The publicly exposed port from the container,
+  if a port mapping exists.
+* `__meta_docker_port_public_ip`: The public IP of the container, if a port
+  mapping exists.
+
+Each discovered container maps into one target per each Docker network it
+belongs to and per each defined port mapping the container has.
```

Comment:
```suggestion
Each discovered container maps to one target per each Docker network it
belongs to and one target per each defined port mapping.
```

Reply:
I'll try to rephrase this in another way to make it more clear, because I think the suggestion here loses some of its meaning. 

If a container uses N docker networks and M defined ports, that container maps into N*M total targets.

Reply:
going for this: 

> Each discovered container maps to one target per unique combination of network
and port mappings used by the container.